lib/ansible/plugins/test/core.py

```diff
@@ -66,6 +67,44 @@ def skipped(result):
     return result.get('skipped', False)
 
 
+def test_boolean(value):
+    ''' Return true if the object is a boolean value '''
+    return value is True or value is False
+
+
+def test_false(value):
+    ''' Return true if the object is False '''
+    return value is False
+
+
+def test_true(value):
+    ''' Return true if the object is True '''
+    return value is True
+
+
+# NOTE: The existing Jinja2 'number' test matches booleans and floats
+def test_integer(value):
+    ''' Return true if the object is an integer '''
+    return isinstance(value, integer_types) and value is not True and value is not False
+
+
+# NOTE: The existing Jinja2 'number' test matches booleans and integers
+def test_float(value):
+    ''' Return true if the object is a float '''
+    return isinstance(value, float)
+
+
+# NOTE: The existing Jinja2 'sequence' test matches strings and dictionaries
+def test_list(value):
+    ''' Return true if the object is a list or tuple '''
+    return isinstance(value, Sequence) and not isinstance(value, string_types)
```

Comment:
JFYI
In the internal slack channel we came up with an idea: https://github.com/ansible/ansible/pull/39924
So it might get into devel before you merge this

Reply:
Well, I think we should be pushing this to Jinja rather than invent our own.
That is why I moved it upstream: pallets/jinja#824

PS I personally don't like the name "is_item_sequence".

Reply:
Honestly, I just failed to invent a better name

Reply:
Having it upstream would be nice, but I doubt we can change our minimum required Jinja2 version anytime soon.

Reply:
@mattclay The idea was to backport it once it is accepted upstream (and the names are set in stone). If we do it now, we risk additional confusion.